pkg/inventory/sanitize.go

```diff
@@ -0,0 +1,35 @@
+package inventory
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+// SanitizeURL takes a raw repository URL and returns only the hostname and path, removing possible
+// prefix protocol, and extension suffixes.
+func SanitizeURL(rawURL string) (string, error) {
+	u, err := url.Parse(rawURL)
+	if err != nil {
+		return "", err
+	}
+
+	urlPath := strings.TrimSuffix(u.EscapedPath(), ".git")
+	return fmt.Sprintf("%s%s", u.Hostname(), urlPath), nil
+}
+
+// CompareURLs compare the informed URLs.
+func CompareURLs(a, b string) bool {
```

Comment:
Should be an easy unit test.

Reply:
Indeed, please consider the latest changes.